packages/sdks/snippets/vue/src/views/blueprints/ProductDetails.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+    <div v-if="!productDetails">
+      <p>Loading product details...</p>
+    </div>
+    <div class="product-details-page" v-else>
+      <h1>{{ productDetails.data?.['name'] }}</h1>
+      <img
+        :src="productDetails.data?.['image']"
+        :alt="productDetails.data?.['name']"
+      />
+      <p>{{ productDetails.data?.['collection'].value.data.copy }}</p>
+      <p>Price: {{ productDetails.data?.['collection'].value.data.price }}</p>
+    </div>
```

Comment:
`productDetails.data?.['name']` syntax is only needed in Angular projects. It is rather ugly so let's not use it if we don't need to?

Also, if you have the `v-if` check on productDetails?.data,  kinda like you did in `packages/sdks/snippets/nuxt/pages/product/category/[handle].vue`, you can remove all the `?.`:

```suggestion
    <div v-if="!productDetails?.data">
      <p>Loading product details...</p>
    </div>
    <div class="product-details-page" v-else>
      <h1>{{ productDetails.data.name }}</h1>
      <img
        :src="productDetails.data.image"
        :alt="productDetails.data.name"
      />
      <p>{{ productDetails.data.collection.value.data.copy }}</p>
      <p>Price: {{ productDetails.data.collection.value.data.price }}</p>
    </div>
```

this is much cleaner overall